ClockDisplay.java

```diff
@@ -17,28 +17,38 @@
     private NumberDisplay hours;
     private NumberDisplay minutes;
     private String displayString;    // simulates the actual display
+    private boolean meridian;
     
     /**
      * Constructor for ClockDisplay objects. This constructor 
      * creates a new clock set at 00:00.
      */
     public ClockDisplay()
     {
-        hours = new NumberDisplay(24);
+        hours = new NumberDisplay(13);
+        if(hours.getValue()==0){
+            hours.setValue(12);
+        }
         minutes = new NumberDisplay(60);
-        updateDisplay();
+        meridian = false;
+        get12HourInternalDisplay();
     }
 
     /**
      * Constructor for ClockDisplay objects. This constructor
      * creates a new clock set at the time specified by the 
      * parameters.
+     * false for AM
+     * true for PM
      */
-    public ClockDisplay(int hour, int minute)
+    public ClockDisplay(int hour, int minute, boolean meridian)
     {
-        hours = new NumberDisplay(24);
+        hours = new NumberDisplay(13);
```

Comment:
internally the number display should only be 0..11 - it is updateDisplay's job to turn that to 1-12